packages/oas/medusa-oas-cli/src/command-oas.ts

```diff
@@ -0,0 +1,148 @@
+
+export const commandOptions: Option[] = [
+  new Option("-t, --type <type>", "API type to compile []")
+    .choices(["all", "admin", "store"])
+    .default("all"),
+  new Option(
+    "-o, --out-dir <outDir>",
+    "Destination directory to output generated OAS files."
+  ).default(process.cwd()),
+  new Option("-D, --dry-run", "Do not output files."),
+  new Option(
+    "-p, --paths <paths...>",
+    "Additional paths to crawl for OAS JSDoc."
+  ),
+  new Option("-F, --force", "Ignore OAS validation and output OAS files."),
+]
+
+export function getCommand() {
+  const command = new Command(commandName)
+  command.description(commandDescription)
+  for (const opt of commandOptions) {
+    command.addOption(opt)
+  }
+  command.action(async (options) => await execute(options))
+  return command
+}
+
+/**
+ * Main
+ */
+export async function execute(cliParams: OptionValues) {
+  /**
+   * Process CLI options
+   */
+  const dryRun = !!cliParams.dryRun
+  const force = !!cliParams.force
+
+  const apiTypesToExport =
+    cliParams.type === "all" ? ["store", "admin"] : [cliParams.type]
+
+  const outDir = path.resolve(cliParams.outDir)
+  if (!(await lstat(outDir)).isDirectory()) {
+    throw new Error(`--outDir must be a directory - ${outDir}`)
+  }
+
+  const additionalPaths = (cliParams.paths ?? []).map((additionalPath) =>
+    path.resolve(additionalPath)
+  )
+  for (const additionalPath of additionalPaths) {
+    if (!(await lstat(additionalPath)).isDirectory()) {
+      throw new Error(`--path must be a directory - ${additionalPath}`)
+    }
+  }
+
+  /**
+   * Command execution
+   */
+  if (!dryRun) {
+    await mkdir(outDir, { recursive: true })
+  }
+
+  for (const apiType of apiTypesToExport) {
+    console.log(`🟣 Generating OAS - ${apiType}`)
+    const oas = await getOASFromCodebase(apiType as ApiType, additionalPaths)
+    await validateOAS(oas, apiType as ApiType, force)
+    if (!dryRun) {
+      await exportOASToJSON(oas, apiType as ApiType, outDir)
+    }
+  }
+}
+
+/**
+ * Methods
+ */
+const getOASFromCodebase = async (
+  apiType: ApiType,
+  additionalPaths: string[] = []
+): Promise<OpenAPIObject> => {
+  const gen = await swaggerInline(
+    [
+      path.resolve(medusaPackagePath, "dist", "models"),
+      path.resolve(medusaPackagePath, "dist", "types"),
+      path.resolve(medusaPackagePath, "dist", "api/middlewares"),
+      path.resolve(medusaPackagePath, "dist", `api/routes/${apiType}`),
+      ...additionalPaths,
+    ],
+    {
+      base: path.resolve(
+        medusaPackagePath,
+        "oas",
+        `${apiType}-spec3-base.yaml`
+      ),
+      format: ".json",
+    }
+  )
+
+  return await OpenAPIParser.parse(JSON.parse(gen))
+}
+
+const validateOAS = async (
+  oas: OpenAPIObject,
+  apiType: ApiType,
+  force = false
+): Promise<void> => {
+  try {
+    await OpenAPIParser.validate(JSON.parse(JSON.stringify(oas)))
+    console.log(`🟢 Valid OAS - ${apiType}`)
```

Comment:
suggestion: for all, should we use the logger instead of console.log? `import logger from "@medusajs/medusa-cli/dist/reporter"`

Reply:
yes you right,it will be better when it will be in its own package. maybe add a todo on that to not forget?